### PR TITLE
Turn editEntities off on domain change

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: interface/src/scripting/WindowScriptingInterface.h [hunk 1]
```diff
@@ -57,6 +57,7 @@ public slots:
     QScriptValue peekNonBlockingFormResult(QScriptValue array);
 
 signals:
+    void domainChanged(const QString& domainHostname);
```
Comment (ZappoMan) [inline interface/src/scripting/WindowScriptingInterface.h]: I thought we already had a signal for this? Is there a common JS way of doing this?

Comment (huffman) [inline interface/src/scripting/WindowScriptingInterface.h]: I know `Application` handles this event, but I am not aware of it being exposed to JS.

I don't know of an in-browser implementation for this in particular.  You can react to a page changing (window.onbeforeunload), but that's a much different context because a script will only run within the context of a single page (not across domains, as we do).

**Comment by hifi-gustavo** [comment]
OS X Yosemite functional test completed with errors
Download recording here:
https://s3.amazonaws.com/hifi-public/pr-builds/recordings/4330.mov

**Comment by hifi-gustavo** [comment]
Build console output for commit: c9818c184a41fb9835e28d58d141c85c8e03b0d5 available at: https://gist.github.com/0664fb3de1e27f5e6c23<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4330.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4330.dmg

File: interface/src/scripting/WindowScriptingInterface.cpp [hunk 2]
```diff
@@ -34,6 +35,8 @@ WindowScriptingInterface::WindowScriptingInterface() :
     _nonBlockingFormActive(false),
     _formResult(QDialog::Rejected) 
 {
+    const DomainHandler& domainHandler = DependencyManager::get<NodeList>()->getDomainHandler();
+    connect(&domainHandler, &DomainHandler::hostnameChanged, this, &WindowScriptingInterface::domainChanged);
```
Comment (jherico) [inline interface/src/scripting/WindowScriptingInterface.cpp]: Does this get triggered only when the domain changes, or also when you disconnect?

Comment (huffman) [inline interface/src/scripting/WindowScriptingInterface.cpp]: Only when the domain changes

Comment (jherico) [inline interface/src/scripting/WindowScriptingInterface.cpp]: Would it not make sense for scripts to be aware of the connected / disconnected state as well?

Comment (huffman) [inline interface/src/scripting/WindowScriptingInterface.cpp]: Yes, I think it would make sense to expose that information, although I think it would be a better fit for a separate PR / feature request.

Comment (ZappoMan) [inline interface/src/scripting/WindowScriptingInterface.cpp]: @jherico - I'm going to merge this PR as is now. But to your point about connect/disconnect. Can you clarify what you mean? We do have a callback for script to know when the script is stopping (usually because the client is shutting down)... do you mean something else by connect/disconnect state? Do you mean the case where if you're connected to a domain, but all of a sudden the domain shutdown even though you didn't leave the domain? Is that the situation you're thinking of?

Comment (jherico) [inline interface/src/scripting/WindowScriptingInterface.cpp]: I didn't have a specific use case in mind, but was mostly thinking about what sorts of states weren't being tracked by only sending a message on hostname changed.  But on reflection I can think of a number of cases.  Right now if I go into my client and say 'disconnect'  (or for any other reason fail to connect to a server) then the window title says 'DISCONNECTED'.  However that kind of window decoration based information will be useless in a VR context where you can't see the window title.  

I can also imagine that a user might want their available UI to change depending on whether they were connected or not, since there are clearly different tasks that will be done in a connected vs a disconnected state.  

I might even go so far to suggest that the concept of a non-server based environment might be desirable.  Consider it an analog to loading local HTML files in a web browser.  However, now we're wildly out of the scope of the this PR.
